src/java.base/share/native/libjli/emessages.h

```diff
@@ -62,60 +62,33 @@
-#define CLS_ERROR3      "Error: No main method found in specified class.\n" GEN_ERROR
-#define CLS_ERROR4      "Error: Main method not public\n" GEN_ERROR
-#define CLS_ERROR5      "Error: main-class: attribute exceeds system limits of %d bytes\n" GEN_ERROR
-
 #define CFG_WARN1       "Warning: %s VM not supported; %s VM will be used"
 #define CFG_WARN2       "Warning: No leading - on line %d of `%s'"
 #define CFG_WARN3       "Warning: Missing VM type on line %d of `%s'"
-#define CFG_WARN4       "Warning: Missing server class VM on line %d of `%s'"
 #define CFG_WARN5       "Warning: Unknown VM type on line %d of `%s'"
 
 #define CFG_ERROR1      "Error: Corrupt jvm.cfg file; cycle in alias list."
 #define CFG_ERROR2      "Error: Unable to resolve VM alias %s"
 #define CFG_ERROR3      "Error: %s VM not supported"
-#define CFG_ERROR4      "Error: Unable to locate JRE meeting specification \"%s\""
 #define CFG_ERROR5      "Error: Could not determine application home."
 #define CFG_ERROR6      "Error: could not open `%s'"
 #define CFG_ERROR7      "Error: no known VMs. (check for corrupt jvm.cfg file)"
-#define CFG_ERROR8      "Error: missing `%s' JVM at `%s'.\nPlease install or use the JRE or JDK that contains these missing components."
+#define CFG_ERROR8      "Error: missing `%s' JVM at `%s'.\nPlease install a JDK that contains these missing components."
 #define CFG_ERROR9      "Error: could not determine JVM type."
 #define CFG_ERROR10     "Error: Argument file size should not be larger than %lu."
 
-#define JRE_ERROR1      "Error: Could not find Java SE Runtime Environment."
-#define JRE_ERROR2      "Error: This Java instance does not support a %d-bit JVM.\nPlease install the desired version."
-#define JRE_ERROR3      "Error: Improper value at line %d."
-#define JRE_ERROR4      "Error: trying to exec %s.\nCheck if file exists and permissions are set correctly."
-#define JRE_ERROR5      "Error: Failed to start a %d-bit JVM process from a %d-bit JVM."
-#define JRE_ERROR6      "Error: Verify all necessary Java SE components have been installed."
-#define JRE_ERROR7      "Error: Either 64-bit processes are not supported by this platform\nor the 64-bit components have not been installed."
-#define JRE_ERROR8      "Error: could not find "
-#define JRE_ERROR9      "Error: Unable to resolve %s"
-#define JRE_ERROR10     "Error: Unable to resolve current executable"
-#define JRE_ERROR11     "Error: Path length exceeds maximum length (PATH_MAX)"
-#define JRE_ERROR12     "Error: Exec of %s failed"
-#define JRE_ERROR13     "Error: String processing operation failed"
+#define LAUNCHER_ERROR1 "Error: Could not find Java SE Runtime Environment."
+#define LAUNCHER_ERROR2 "Error: could not find "
+#define LAUNCHER_ERROR3 "Error: Path length exceeds maximum length (PATH_MAX)"
+#define LAUNCHER_ERROR4 "Error: trying to exec %s.\nCheck if file exists and permissions are set correctly."
+#define LAUNCHER_ERROR5 "Error: String processing operation failed"
```

Comment:
While renaming these `JRE_ERROR` macros, I've renumbered them, since some of these `JRE_ERROR<N>` were unused. Relevant usages within the code of these macros have been updated accordingly to make sure they continue to emit the correct message.